actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -14,10 +14,17 @@ module MimeNegotiation
       #
       # For backward compatibility, the post \format is extracted from the
       # X-Post-Data-Format HTTP header if present.
-      def content_mime_type
+      def content_mime_type(validate: true)
         fetch_header("action_dispatch.request.content_type") do |k|
           v = if get_header("CONTENT_TYPE") =~ /^([^,\;]*)/
-            Mime::Type.lookup($1.strip.downcase)
+            Mime::Type.lookup($1.strip.downcase).tap do |content_mime_type|
+              next unless validate
+              # When the content type is known, Mime::Type.lookup returns <tt>Mime::Type::LOOKUP[string]</tt>.
+              # Otherwise, it returns <tt>Mime::Type.new(string)</tt> which has a the symbol +nil+.
+              next unless content_mime_type.symbol.nil?
+              next if env.key?("action_dispatch.exception") # Don't raise in exceptions_app, else FAILSAFE_RESPONSE
+              raise ActionController::UnsupportedMediaType.new(content_mime_type)
```

Comment:
@rafaelfranca I implemented raising the 415 as requested when getting the `content_mime_type`, but since there's a test that checks for a form post media type that doesn't expect an exception, I added the `validate: true` default.  I don't really like it... needs thought 86c7c391fce919ac6a547887174c7257d43ff002